client/handle_http.go

```diff
@@ -2607,31 +2679,36 @@ func statHttp(ctx context.Context, dest *url.URL, namespace namespaces.Namespace
 			canDisableProxy := CanDisableProxy()
 			disableProxy := !isProxyEnabled()
 
-			var resp *http.Response
+			var info fileInfo
 			for {
 				if disableProxy {
-					log.Debugln("Performing HEAD (without proxy)", endpoint.String())
+					log.Debugln("Performing request (without proxy)", endpoint.String())
 					transport.Proxy = nil
 				} else {
-					log.Debugln("Performing HEAD", endpoint.String())
-				}
-
-				var req *http.Request
-				req, err = http.NewRequestWithContext(ctx, http.MethodHead, endpoint.String(), nil)
-				if err != nil {
-					log.Errorln("Failed to create HTTP request:", err)
-					resultsChan <- statResults{0, err}
-					return
-				}
-
-				if token != "" {
-					req.Header.Set("Authorization", "Bearer "+token)
+					log.Debugln("Performing request", endpoint.String())
 				}
+				client.SetTransport(transport)
 
-				resp, err = client.Do(req)
+				fsinfo, err := client.Stat(endpoint.Path)
 				if err == nil {
+					info = fileInfo{
+						Name:    path.Base(endpoint.Path),
+						Size:    fsinfo.Size(),
+						IsDir:   fsinfo.IsDir(),
```

Comment:
What does this look like outside of POSIX origins? If I were to use `client.Stat` against an S3 prefix, do we say "IsDir: true", or is that behavior undefined?

It's fine if we don't have an answer now, but something we should be asking.